src/addrman.cpp

```diff
@@ -188,7 +188,7 @@ void AddrManImpl::Serialize(Stream& s_) const
 
     int nUBuckets = ADDRMAN_NEW_BUCKET_COUNT ^ (1 << 30);
     s << nUBuckets;
-    std::unordered_map<int, int> mapUnkIds;
+    std::unordered_map<nid_type, int> mapUnkIds;
```

Comment:
Hmm, can you explain the "morally" part a bit more?

`mapUnkIds` maps `nid_type`  to an `int`, let's call that counter `ser_count`. `ser_count` is part of the serialization and is bounded by the limited slots of the addrman tables - even if the internal id would overflow an `int`, `ser_count` wouldn't be affected by that.

During deserialization, `ser_count` is read into `int entry_index` (which must be `int`), and then makes it into `bucket_entries`  and later into `vvNew`. But I'm unsure when is the best step to convert it into an `int64_t` -  why is it better to already do that in the step `entry_index` -> `bucket_entries` rather than the way it is now?

Reply:
I'm fine with the version on the branch now because `Serialise()` and `Unserialise()` are symmetrical.  `nIndex` is [written as int](https://github.com/bitcoin/bitcoin/blob/51f7668d31e2624e41c7ce77fe33162802808f3f/src/addrman.cpp#L220) into peers.dat in `Serialise()` and [read as int](https://github.com/bitcoin/bitcoin/blob/51f7668d31e2624e41c7ce77fe33162802808f3f/src/addrman.cpp#L325) from peers.dat in `Unserialise()`.